File: 2026-03-10-oop/07-object-initialization/src/car.ts
export default class Car {
    // members
    make: string
    engineVolume: number
    horsePower: number
    color: string
    model: number

    // methods 
    drive() {

        // i need a way to refer to an actual object
        // instansiated from this class, within the class
        // the way to do it, is by using the word: this

        console.log(`${this.make} voom voom`)
    }

    break() {
        console.log(`${this.make} heheheheyyyyywwww`)
    }

    ignite() {
        console.log(`${this.make} phphphphphph`)
    }

    constructor (make: string, engineVolume: number, horsePower: number, color: string, model: number) {
    // constructor (make: string, engineVolume: number) {
        this.make = make
        this.engineVolume = engineVolume
        this.horsePower = horsePower
        this.color = color
        this.model = model
    }
}

File: 2026-08-04-python/01-homework-1/car.py
class Car:
    def __init__(self: Car, make: str, model: str, color: str, price: float):
        self.make = make
        self.model = model
        self.color = color
        self.price = price

    def display(self):
        print(f"Make: {self.make}")
        print(f"Model: {self.model}")
        print(f"Color: {self.color}")
        print(f"Price: {self.price}")

    @property
    def brand(self):
        return f"{self.make} {self.model}"

